pandas/core/groupby/groupby.py

```diff
@@ -318,14 +318,13 @@ class providing the base-class of operations.
 """
 
 _transform_template = """
-Call function producing a like-indexed %(klass)s on each group and
-return a %(klass)s having the same indexes as the original object
-filled with the transformed values
+Apply function ``func`` column-by-column to the GroupBy object and return a %(klass)s
```

Comment:
This is sometimes true, but not always. For example:

```
def foo(x):
    print(x)
    return x

df = pd.DataFrame({'a': [1, 2], 'b': [2, 3], 'c': [3, 4]})
df.groupby('a').transform(foo)
```

gives (after clipping of the first group which is used to determine slowpath vs fastpath)

```
   b  c
0  2  3
   b  c
1  3  4
```

In this case, the transform is evaluated on the entire group, not column-by-column.

Reply:
My recommendation here would be to combine the previous version (calling on the group) along with what you have (column-by-column). If calling on the first group is successful, then transform with operate group-by-group. Otherwise, it falls back to column-by-column (within each group).

Reply:
Thanks for the input! However, I think combining the grouping + columns is a bit confusing. How would a user differentiate between using the .apply() vs the .transform() function then?

Reply:
@willie3838 - Agreed this is a part of the API that needs cleaning up, but I'd recommend leaving that aspect out of scope for this PR and documentation the behavior as it currently exists. If you agree with my assessment that the current documentation in this PR is not correct, then it needs to be fixed. On the other hand, if you think my assessment is not right, then let me know how!